diff --git a/CMakeLists.txt b/CMakeLists.txt
index db6c035..69a4e67 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -7,7 +7,7 @@ include(CheckLibraryExists)
 find_package(ament_cmake REQUIRED)
 find_package(ament_cmake_ros_core REQUIRED)
 
-if(UNIX AND NOT APPLE)
+if(UNIX AND NOT APPLE AND NOT QNX)
   include(cmake/check_c_compiler_uses_glibc.cmake)
   check_c_compiler_uses_glibc(USES_GLIBC)
   if(USES_GLIBC)
diff --git a/cmake/check_c_compiler_uses_glibc.cmake b/cmake/check_c_compiler_uses_glibc.cmake
index 46755be..0c778c2 100644
--- a/cmake/check_c_compiler_uses_glibc.cmake
+++ b/cmake/check_c_compiler_uses_glibc.cmake
@@ -32,5 +32,5 @@ function(check_c_compiler_uses_glibc result_variable)
   ]====])
 
   check_c_source_compiles("${GLIBC_TEST_CODE}" ${result_variable})
-  set(${result_variable} ${result_variable} PARENT_SCOPE)
+  set(${result_variable} ${${result_variable}} PARENT_SCOPE)
 endfunction()
diff --git a/src/find.c b/src/find.c
index d5dbf7a..a73f471 100644
--- a/src/find.c
+++ b/src/find.c
@@ -60,7 +60,7 @@ rcutils_find_lastn(const char * str, char delimiter, size_t string_length)
     return SIZE_MAX;
   }
 
-#if defined(_GNU_SOURCE)
+#if defined(_GNU_SOURCE) && !defined(__QNXNTO__) && !defined(__QNX__)
   const char * ptr = memrchr(str, delimiter, string_length);
   if (ptr == NULL) {
     return SIZE_MAX;
